doc/sphinx/proof-engine/tactics.rst

```diff
@@ -1506,160 +1494,140 @@ Managing the local context
-   of name :n:`H` asserting :g:`U` to the current goal and opens a new subgoal
-   :g:`U` [2]_. The subgoal :g:`U` comes first in the list of subgoals remaining to
-   prove.
-
-   .. exn:: Not a proposition or a type.
-
-      Arises when the argument :token:`type` is neither of type :g:`Prop`,
-      :g:`Set` nor :g:`Type`.
-
-   .. tacv:: assert @type
-
-      This behaves as :n:`assert (@ident : @type)` but :n:`@ident` is
-      generated by Coq.
-
-   .. tacv:: assert @type by @tactic
-
-      This tactic behaves like :tacn:`assert` but applies tactic to solve the
-      subgoals generated by assert.
-
-      .. exn:: Proof is not complete.
-         :name: Proof is not complete. (assert)
-         :undocumented:
-
-   .. tacv:: assert @type as @simple_intropattern
-
-      If :n:`simple_intropattern` is an intro pattern (see :ref:`intropatterns`),
-      the hypothesis is named after this introduction pattern (in particular, if
-      :n:`simple_intropattern` is :n:`@ident`, the tactic behaves like
-      :n:`assert (@ident : @type)`). If :n:`simple_intropattern` is an action
-      introduction pattern, the tactic behaves like :n:`assert @type` followed by
-      the action done by this introduction pattern.
-
-   .. tacv:: assert @type as @simple_intropattern by @tactic
+.. tacn:: assert ( @ident : @type ) {? by @ltac_expr3 }
+          assert ( @ident := @term )
+          assert @one_term {? @as_ipat } {? by @ltac_expr3 }
+   :name: assert; _; _
 
-      This combines the two previous variants of :tacn:`assert`.
+   Adds a new hypothesis to the current subgoal and a
+   new subgoal to prove the hypothesis.  The new subgoal becomes
+   the first in the list of subgoals.  Then, if :n:`@ltac_expr3`
+   is specified, it applies that tactic to fully prove the new subgoal (and
+   otherwise fails).
 
-   .. tacv:: assert (@ident := @term)
+   The first form adds a new hypothesis named :n:`@ident` of type :n:`@type`.
+   (This corresponds to the cut rule of sequent calculus.)
 
-      This behaves as :n:`assert (@ident : @type) by exact @term` where
-      :token:`type` is the type of :token:`term`. This is equivalent to using
-      :tacn:`pose proof`. If the head of term is :token:`ident`, the tactic
-      behaves as :tacn:`specialize`.
+   The second form is equivalent to :n:`assert (@ident : @type) by exact (@term)` where
+   :n:`@type` is the type of :n:`@term`.  It is also equivalent to using
+   :tacn:`pose proof`. If the head of :n:`@term` is :n:`@ident`, the tactic
+   is equivalent to :tacn:`specialize`.
 
-      .. exn:: Variable @ident is already declared.
-         :undocumented:
-
-.. tacv:: eassert @type as @simple_intropattern by @tactic
-   :name: eassert
+   In the third form, Coq chooses fresh name for the hypothesis if :n:`@as_ipat`
+   isn't specified.  Otherwise, it transforms the type
+   of :n:`@term` as specified by :n:`@as_ipat` and adds the resulting new hypotheses.
```

Comment:
```suggestion
   isn't specified.  Otherwise, it transforms the hypothesis as specified by :n:`@as_ipat` and adds the resulting new hypotheses.
```

There is no "type of term" that should be discussed here. `one_term` is the hypothesis (see my comment above).